src/server.js

```diff
@@ -354,3 +367,69 @@ function auth(data) {
 		}
 	}
 }
+
+function sendConnectionInfo(client) {
+	// send messages to connected user clients
+	// get sockets
+	var socketList = Object.keys(client.attachedClients);
+
+	// get IP addresses for those sockets
+	const connectionList = socketList.map(socket => ({
+		socketId: socket,
+		ip: getClientIp(manager.sockets.of("/").connected[socket].request)
+	}));
+
+	// get hostnames for every ip
+	hostnamePromise(connectionList)
+		.then(function(list) {
+			// send event for every connection
+			for (let j = 0; j < list.length; j++) {
+				// make data packet for event
+				let sendData = {
+					currentSession: null,
+					otherSessions: []
+				};
+				// compile data for every connection
+				for (let i = 0; i < list.length; i++) {
+					let data = {};
+					data.hostname = list[i].hostname;
+					data.ip = list[i].ip;
+					data.socketId = list[i].socketId;
+					// active host
+					if (list[j].socketId === list[i].socketId) {
+						sendData.currentSession = data;
+					} else {
+						sendData.otherSessions.push(data);
+					}
+				}
+				// send event with data
+				manager.sockets.of("/")
+					.connected[list[j].socketId]
+					.emit("update-clients-list", sendData);
+			}
+		})
+		.catch(function(err) {
+			log.error("Unable to resolve hostname", err);
+		});
+}
+
+function hostnamePromise(data) {
+	if (Array.isArray(data)) {
```

Comment:
My idea was, that i allow `hostnamePromise("192.168.1.1")` and `hostnamePromise(["192.168.1.1"])`. One way or other, you will need `IsArray` check.